admin_scripts/aula_register.py

```diff
@@ -0,0 +1,205 @@
+try:
+    browser.get(url)
+except InvalidArgumentException as iae:
+    print('Invalid argument given: {} of type {}'.format(url, type(url)))
+    print(iae.message)
+    sys.exit(1)
+
+print('Chrome browser opened headless with url: {}'.format(url))
+register_code_list = list(register_code)
+wait.until(expected.visibility_of_element_located((By.id, '__BVID__11'))).send_keys('' + register_code_list[0] + Keys.ENTER)
+wait.until(expected.visibility_of_element_located((By.id, '__BVID__12'))).send_keys('' + register_code_list[1] + Keys.ENTER)
+wait.until(expected.visibility_of_element_located((By.id, '__BVID__13'))).send_keys('' + register_code_list[2] + Keys.ENTER)
+wait.until(expected.visibility_of_element_located((By.id, '__BVID__14'))).send_keys('' + register_code_list[3] + Keys.ENTER)
+wait.until(expected.visibility_of_element_located((By.id, '__BVID__15'))).send_keys('' + register_code_list[4] + Keys.ENTER)
+wait.until(expected.visibility_of_element_located((By.id, '__BVID__16'))).send_keys('' + register_code_list[5] + Keys.ENTER)
+wait.until(expected.visibility_of_element_located((By.id, '__BVID__17'))).send_keys('' + register_code_list[6] + Keys.ENTER)
+wait.until(expected.visibility_of_element_located((By.id, '__BVID__18'))).send_keys('' + register_code_list[7] + Keys.ENTER)
+
+try:
+    wait.until(expected.visibility_of_element_located((By.CLASS_NAME, 'aula-presence-overview')))
+except:
+    print('Aula presence overview did not show up. Check that the register code is correct and valid: {}'.format(register_code))
+    sys.exit(1)
+
+cookies_list = browser.get_cookies()
+print('Cookies {}'.format(cookies_list))
+
+browser.close()
+print('Google Chrome headless browser closed.')
+
+chrome_db_path = '/home/.skjult/.config/google-chrome/Default/Cookies'
+conn = None
+try:
+    conn = sqlite3.connect(chrome_db_path)
+except Error as e:
+    print('DB connection could not be established on path {}'.format(chrome_db_path))
+    sys.exit(1)
+
+cursor = conn.cursor()
+
+# .skjult can contain old cookies table with old table structure.
+cursor.execute("SELECT sql FROM sqlite_master WHERE name='cookies'")
+rows = cursor.fetchall()
+# Try to detect old cookies table structure...
+if 'is_secure' not in rows[0][0]:
+    # if cookies table is out of date drop it and create it with new structure.
+    print('Updating cookies table...')
+    cursor.execute("DROP TABLE main.cookies")
+    conn.commit()
+    cursor.execute("CREATE TABLE cookies(creation_utc INTEGER NOT NULL,host_key TEXT NOT NULL,name TEXT NOT NULL,value TEXT NOT NULL,path TEXT NOT NULL,expires_utc INTEGER NOT NULL,is_secure INTEGER NOT NULL,is_httponly INTEGER NOT NULL,last_access_utc INTEGER NOT NULL,has_expires INTEGER NOT NULL DEFAULT 1,is_persistent INTEGER NOT NULL DEFAULT 1,priority INTEGER NOT NULL DEFAULT 1,encrypted_value BLOB DEFAULT '',samesite INTEGER NOT NULL DEFAULT -1,UNIQUE (host_key, name, path))")
+    conn.commit()
+    print('Cookies table has been updated.')
+else:
```

Comment:
![emoji_u1f635](https://user-images.githubusercontent.com/45562185/68573133-61f7e500-0467-11ea-8b9b-6dc27350dea8.png)

What does this code do? Is it responsible for converting the ancient cookie database in `/home/.skjult` into something fractionally more modern?

Reply:
Yes :)
I could not come up with a better solution...

Reply:
_Ghrhghjghjk._

(I wonder if it's possible to make this less... brittle. For example, could we perhaps make a Chrome extension that turns a preference value into a persistent cookie?)

Reply:
That would be nice... It would also make the script less vunerable for table changes in the future.